### ISSUE TITLE
🐛 Bug: Browser doesn't include call stack for global uncaught errors

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [mochajs/mocha#5106]
### Bug Report Checklist

- [X] I have read and agree to Mocha's [Code of Conduct](https://github.com/mochajs/mocha/blob/master/.github/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/mochajs/mocha/blob/master/.github/CONTRIBUTING.md)
- [X] I have searched for [related issues](https://github.com/mochajs/mocha/issues?q=is%3Aissue) and [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20), but none matched my issue.
- [X] I have 'smoke tested' the code to be tested by running it outside the real test suite to get a better sense of whether the problem is in the code under test, my usage of Mocha, or Mocha itself.
- [X] I want to provide a PR to resolve this

### Expected

When an uncaught error occurs during a test in browsers, you'd expect to see the call stack. That's useful.

```plaintext
Error: Oh no!
    at inner (test.example.js:8:15)
```

### Actual

Mocha's browser error handling code was written before browsers gave nice stacks (https://github.com/mochajs/mocha/commit/eb58761bb3346d2597e8365e60273417127b72c4).

```plaintext
Error: Error: Oh no! (http://127.0.0.1:8081/test.example.js:8)
```

### Minimal, Reproducible Example

Using a minimal browser setup such as in https://github.com/mochajs/mocha-examples/pull/72:

```js
it("example", (done) => {
    function inner() {
        throw new Error("Oh no!");
    }

    setTimeout(inner);
});
```

### Versions

Mocha: 10.3.0

### Additional Info

This was filed long ago in #2167, then a fix proposed in #3952. Since #2167 is quite old, re-filing to be more current.

**Comment by JoshuaKGoldberg** [comment]
Ah that's nifty to hear you're using Mocha in the Chrome DevTools! We're just waiting for review from another maintainer on #5107. I know everyone else has been swamped lately, so unclear when we'll be able to.

**Comment by JoshuaKGoldberg** [comment]
#5107 is merged; we'll release a new version with the fix soon. 🚀

**Comment by JoshuaKGoldberg** [comment]
#5107 was released in [`mocha@10.5.0`](https://www.npmjs.com/package/mocha/v/10.5.0). 🚀

**Comment by ergunsh** [comment]
Hey @JoshuaKGoldberg, sorry for the late update on my side. Thanks a lot for your contribution here and for the release as well. We started using the new version as of last month and happy to see that the issue is fixed! :)
